### PR TITLE
GetDateTimeOffset and fix pageSizeHint for Pageable

### LINKED ISSUE #13921: Expose pagination of results through `Pageable.AsPages`
C# typically exposes `maxPerPage` through `.AsPages(int? pageSizeHint)`, so[ look into pagination in other SDKs](https://github.com/Azure/azure-sdk-for-net/pull/13824#discussion_r463718291) to see the best way to implement that.

## Resources
- https://docs.microsoft.com/en-us/dotnet/api/azure.pageable-1?view=azure-dotnet
- https://github.com/meanjeekim/azure-sdk-for-net/blob/master/sdk/search/Azure.Search.Documents/src/Models/SearchResults%7BT%7D.cs#L325

### LINKED ISSUE #14002: Implement GetDateTimeOffset in TableEntity

### PR DESCRIPTION
**Pull request by christothes**
fixes #14002
fixes #13921

### CODE REVIEW

File: sdk/tables/Azure.Data.Tables/tests/TableServiceClientLiveTests.cs [hunk 1]
```diff
@@ -175,12 +174,23 @@ public async Task GetTablesReturnsTablesWithAndWithoutPagination(int? pageCount)
...
-
-                Assert.That(() => tableResponses, Is.Not.Empty);
-                Assert.That(() => tableResponses.Select(r => r.TableName), Contains.Item(tableName));
+                var remainingItems = createdTables.Count;
+                await foreach (var page in service.GetTablesAsync(/*maxPerPage: pageCount*/).AsPages(pageSizeHint: pageCount))
+                {
+                    Assert.That(page.Values, Is.Not.Empty);
+                    if (pageCount.HasValue)
+                    {
+                        Assert.That(page.Values.Count, Is.EqualTo(Math.Min(pageCount.Value, remainingItems)));
```
Comment (pakrym) [inline sdk/tables/Azure.Data.Tables/tests/TableServiceClientLiveTests.cs:183]: LessThanOrEqualTo?

File: sdk/tables/Azure.Data.Tables/src/TableServiceClient.cs [hunk 2]
```diff
@@ -221,21 +221,24 @@ public virtual AsyncPageable<TableItem> GetTablesAsync(string filter = null, int
...
-                       new QueryOptions() { Filter = filter, Select = null, Top = maxPerPage, Format = _format },
-                       cancellationToken).ConfigureAwait(false);
-                return Page.FromValues(response.Value.Value, response.Headers.XMsContinuationNextTableName, response.GetRawResponse());
-            });
+                return PageableHelpers.CreateAsyncEnumerable(
+                    async pageSizeHint =>
+                    {
+                        var response = await _tableOperations.QueryAsync(
+                            null,
+                            new QueryOptions() { Filter = filter, Select = null, Top = pageSizeHint, Format = _format },
```
Comment (pakrym) [inline sdk/tables/Azure.Data.Tables/src/TableServiceClient.cs:229]: Extracting this value might make the code read a bit cleaner.

File: sdk/tables/Azure.Data.Tables/tests/TableServiceClientLiveTests.cs (hunk 1 above)
Comment (christothes) [reply sdk/tables/Azure.Data.Tables/tests/TableServiceClientLiveTests.cs:183]: Isn't that less precise than what is there now?

**Review by pakrym — APPROVED** [review]

File: sdk/tables/Azure.Data.Tables/tests/TableServiceClientLiveTests.cs (hunk 1 above)
Comment (pakrym) [reply sdk/tables/Azure.Data.Tables/tests/TableServiceClientLiveTests.cs:183]: True.

File: sdk/tables/Azure.Data.Tables/src/TableServiceClient.cs (hunk 2 above)
Comment (pakrym) [reply sdk/tables/Azure.Data.Tables/src/TableServiceClient.cs:229]: I meant in the outer scope, so you can define it once and reuse it.

**Pull request merged**